pkg/agent/controller/networkpolicy/fqdn.go

```diff
@@ -482,20 +482,18 @@ func (f *fqdnController) onDNSResponseMsg(dnsMsg *dns.Msg, lookupTime time.Time,
 // wait for another attempt of realization of these rules, before forwarding the response to the
 // original client.
 func (f *fqdnController) syncDirtyRules(fqdn string, waitCh chan error, addressUpdate bool) {
+	dirtyRules := sets.New[string]()
```

Comment:
When waitCh is nil and addressUpdate is false, previously it didn't need to calculate dirtyRules, now it would calculate it unnecessarily.

Reply:
Fixed.